honeybee_3dm/config.py

```diff
@@ -0,0 +1,432 @@
+
+    include_child_layers: bool = Field(
+        False,
+        description='Boolean to indicate if objects from child layers are to be imported'
+    )
+
+    radiance_material: str = Field(
+        None,
+        description='Text string of the radiance identifier.'
+    )
+
+    grid_settings: GridSettings = Field(
+        None
+    )
+
+    honeybee_face_object: FaceObjects = Field(
+        None,
+        description='A text string for FaceObject to create either a Honeybee Shade,'
+        ' a Honeybee door, or a Honeybee aperture object.'
+    )
+
+    honeybee_face_type: FaceType = Field(
+        None,
+        description='A text string for FaceType to create either a Honeybee Wall,'
+        ' a Honeybee Floor, or a Honeybee aperture roof/ceiling, or a Honeybee.'
+        ' Floor.'
+    )
+
+
+class Config(BaseModel):
+    """Config for the config file."""
+
+    sources: Dict[str, str] = Field(
+        None,
+        description='Path to the radiance .mat file.'
+    )
+
+    layers: Dict[str, LayerConfig] = Field(
+        description='names of layer in rhino.'
+    )
+
+    @validator('sources')
+    def check_sources(cls, v):
+        sources = list(v.keys())
+        if len(sources) != 1:
+            raise ValueError('sources can only have one key.')
+
+        if sources[0] != 'radiance_material':
+            raise ValueError(
+                    f'invalid sources key: {sources[0]}.'
+                    'key must be radiance_material'
+            )
+        return v
+    
+    @validator('layers')
+    def check_layers(cls, v):
+        layers = list(v.keys())
+        if not layers:
+            raise ValueError('The config file cannot be used without layer names.')
+        return v
+
+
+def read_json(config_path):
+    """Get a dictionary from a config.json file.
+
+    Args:
+        path: A text string for the path to the config.json.
+
+    Returns:
+        A dictionary for the config.json
+    """
+    try:
+        with open(config_path) as fh:
+            config = json.load(fh)
+    except json.decoder.JSONDecodeError:
+        raise ValueError(
+            'Not a valid config file.'
+        )
+    return config
+
+
+def check_layers(file_3dm, config):
+    """Checks if layers in the config file are layers from the rhino file.
+
+    Args:
+        file_3dm: A rhino3dm file object.
+        config: Config dictionary.
+
+    Raises:
+        KeyError: If any of the layer names mentioned in the config file is not found
+            in the layers in the rhino file.
+
+    Returns:
+        Boolean value of True if no error is raised.
+    """
+    layers = [layer.Name for layer in file_3dm.Layers]
+    layer_check = [True for layer in config['layers'] if layer in layers]
+    if len(config['layers']) != layer_check.count(True):
+        raise KeyError(
+            'Only layer names from the Rhino file are allowed in the config file.'
+        )
+    else:
+        return True
+
+
+def check_rad(config):
```

Comment:
Since this check only has one input which is config - I think this is a better fit to be added to Config as a validator.

Reply:
The way I see it, the validators work on fields. This function checks both the radiance path in `sources` and layer keys in `layers`. It therefore works with two different fields on two different objects (Config and LayerConfig). Please also note that a value (`modifiers_dict`) is passed between two checks in this function.
It seemed complicated to break this function into two validators for two different objects. So I decided to keep as is. Please share if this is unacceptable and I will work on this.

Reply:
> The way I see it, the validators work on fields.

That's not really correct. We use cross reference checks all over the place and it is one of the strengths of Pydantic. See how the `values` are passed after `v` in the first example for validators: https://pydantic-docs.helpmanual.io/usage/validators/ It is called `passwords_match`.

There is also a `root_validator` object: https://pydantic-docs.helpmanual.io/usage/validators/#root-validators but this case is simple enough to just use the `@validator` with `values` argument.

Reply:
Complied